homeassistant/components/sensor/__init__.py

```diff
@@ -271,15 +271,18 @@ def device_class(self) -> SensorDeviceClass | None:
     @property
     def _numeric_state_expected(self) -> bool:
         """Return true if the sensor must be numeric."""
+        device_class = try_parse_enum(SensorDeviceClass, self.device_class)
+        if device_class in {*NON_NUMERIC_DEVICE_CLASSES}:
```

Comment:
You don't need to make a copy if you're not adding `None` as a valid option. 
```suggestion
        if device_class in NON_NUMERIC_DEVICE_CLASSES:
```